### DISCUSSION TITLE
Parsing DNS data error

### DISCUSSION THREAD

**Discussion by sleepingbel** [Security-Onion-Solutions/securityonion#2137] [Unsupported Versions] [answered]
Hello all,

SO version 2.3.10
8 Cores
16 Gb Ram
2.5 Tb disk space
suricata

Could it be that the parsing of DNS packets is not correct ? This is in the standard SO Kibana dashboard DNS in the Network dashboard

In the kibana logs I see that the DNS request and answer packet have the same source and destination ip 

![afbeelding](https://user-images.githubusercontent.com/29328171/100655927-44383f00-334c-11eb-8a8c-d2c132235384.png)
![afbeelding](https://user-images.githubusercontent.com/29328171/100656276-b872e280-334c-11eb-8e28-7d44803fcf8c.png)

This is the PCAP screen
![afbeelding](https://user-images.githubusercontent.com/29328171/100656062-6b8f0c00-334c-11eb-8637-c06f2d634c7f.png)

All the DNS packets in kibana are like this.

I've never made a parse script before, so I don't know what's going wrong in the standard parse script.

Regards
Bart

**Comment by dougburks** [comment, marked as answer]
Here is the parser for Suricata DNS logs:
https://github.com/Security-Onion-Solutions/securityonion/blob/master/salt/elasticsearch/files/ingest/suricata.dns

As you can see, we don't do anything special to source IP or destination IP when parsing. 

I would think that Suricata is intentionally logging the DNS client as the source IP and the DNS server as the destination IP.

**Reply by jasonish** [reply]
Correct.  A DNS request/response will result in 2 events.  In both events, the src_ip will be that of the client, and dest_ip will be that of the server. Most, if not all Suricata application layer logging works this way.

**Reply by sleepingbel** [reply]
Ok that explains a lot.

Thanks now I can experiment further with writing parsers for syslog logs

Doug and Jason thanks for the info
